src/cps/loader.cpp

```diff
@@ -87,7 +87,7 @@ namespace cps::loader {
             if (str == "symbolic") {
                 return Type::symbolic;
             }
-            CPS_UNREACHABLE(fmt::format("Unknown component.type: `{}`", str).c_str());
+            return Type::unknown;
```

Comment:
In retrospect this was an inappropriate place to use `CPS_UNREACHABLE`, it makes more sense in the case of the `enum -> string`, where we actually can have high confidence that the input is something expected and that hitting in the URNEACHABLE case is a developer bug.

That said, I'm somewhat uncomfortable with adding an enum type that is essentially an error type, my preference would be to either use an `expected` or `optional` to note that something has gone wrong.

Reply:
I went back and forth on this until I read these lines from the spec:

> If the type is not recognized by the parser, the component shall be ignored. (Parsers are permitted to support additional types as a conforming extension.)

This is not an error as much as it an unrecognized type by this parser. The set of valid types is open.

Reply:
Side bar on enum/switch/return. I've debated myself about whether unreachable is even a good idea in the enum-to-string case. In C++, it is trivial to construct an enum from an integer. I've had instances where I used this pattern, which led to confusing bugs. In cases where the switch/return is over an enum, I either abort at the end or change the function to return optional and make the function consumer choose how much they trust the input.

Reply:
Okay, with that escape hatch of "additional types are allowed" then that seems reasonable.